src/Control/Monad/Trans/Compose.hs

```diff
@@ -36,10 +37,6 @@ import Data.Foldable (Foldable(fold, foldMap, foldr, foldl, foldr1, foldl1))
 import Data.Traversable (Traversable(traverse, sequenceA, mapM, sequence))
 import Prelude hiding (foldr, foldl, foldr1, foldl1, mapM, sequence)
 
-#if !MIN_VERSION_base(4,11,0)
-import Control.Monad.Fail (MonadFail(..))
-#endif
-
 infixr 9 `ComposeT`
```

Comment:
tldr:
```haskell
#if !MIN_VERSION_base(4,13,0)
```
Is default, and exactly `bases` reported from downstream.

---

* From: `MonadFail` class import can be used from `base 4.9.0.0` on, which is `GHC 8.0`, where `MonadFail` class was added.
* `base 4.11.0.0` (`GHC 8.4.1`) added `instance for Control.Monad.Strict.ST.ST`.
* To: `MonadFail` import should be used until `base 4.13.0.0` (`GHC 8.8`), where `fail` was removed and `fail` of `MonadFail` got exported instead.

And your current patch means that code supports the import of `MonadFail`: ` base ( 4.11.0.0 .. 4.9.0.0 ] `, ` GHC ( 8.4.1 .. 8.0 ]`.

Citing `ddssff` report:
> fails on 8.4.4 and 8.6.5 ...

8.6.5 is `4.12.x`, you used `< 4.11` so the bound probably should be bumped to `< 4.13`.

So, since `!` inverts inclusion, and so means `<`:
```haskell
#if !MIN_VERSION_base(4,13,0)
```
Which is in fact what everybody use for it, since after the `8.6.5` the `8.8` where the import happens automatically.

Personally, if there is no `7.0` support, I just lazily use:
```haskell
#if __GLASGOW_HASKELL__ < 880
import           Prelude hiding (fail)
import           Control.Monad.Fail
#endif
```

It is easier to remember for me.